room/room-runtime/src/main/java/androidx/room/RoomOpenHelper.java

```diff
@@ -153,7 +153,8 @@ private void checkIdentity(SupportSQLiteDatabase db) {
             if (!mIdentityHash.equals(identityHash) && !mLegacyHash.equals(identityHash)) {
                 throw new IllegalStateException("Room cannot verify the data integrity. Looks like"
                         + " you've changed schema but forgot to update the version number. You can"
-                        + " simply fix this by increasing the version number.");
+                        + " simply fix this by increasing the version number. Current identity: "
+                        + identityHash + ", old: " + mIdentityHash + ", legacy: " + mLegacyHash);
```

Comment:
I would remove the `legacy` hash since it is not helpful for those in Room 2.x+ and its flawed, so might add to the confusion.

Moreover, the words 'current' and 'old' are a bit weird here depending on how it's interpreted, for me the hashes are more like 'expected' and 'found'. Where the expected hash is based on your current analyzed schema, while the found one is the one in the database.

Reply:
Great point. Will address.